DigitalLearningSolutions.Data.Tests/Services/GroupServiceTests/EnrolDelegateTests.cs

```diff
@@ -0,0 +1,667 @@
+                reusableDelegateDetails,
+                reusableAccountDetailsData,
+                1,
+                "baseUrl"
+            );
+
+            // Then
+            using (new AssertionScope())
+            {
+                NewDelegateProgressRecordMustNotHaveBeenAdded();
+                A.CallTo(
+                    () => progressDataService.UpdateProgressSupervisorAndCompleteByDate(
+                        reusableProgressRecord.ProgressId,
+                        A<int>._,
+                        null
+                    )
+                ).MustHaveHappened();
+            }
+        }
+
+        [Test]
+        public void
+            EnrolDelegateOnGroupCourses_update_sets_CompleteByDate_as_correct_future_date_if_course_CompleteWithinMonths_is_not_zero()
+        {
+            // Given
+            const int completeWithinMonths = 3;
+            var groupCourse = GroupTestHelper.GetDefaultGroupCourse(completeWithinMonths: completeWithinMonths);
+            var expectedFutureDate = testDate.AddMonths(3);
+            SetupEnrolProcessFakes(
+                GenericNewProgressId,
+                GenericRelatedTutorialId,
+                groupCourse,
+                reusableProgressRecord
+            );
+
+            // When
+            groupsService.EnrolDelegateOnGroupCourses(
+                reusableDelegateDetails,
+                reusableAccountDetailsData,
+                1,
+                "baseUrl"
+            );
+
+            // Then
+            using (new AssertionScope())
+            {
+                NewDelegateProgressRecordMustNotHaveBeenAdded();
+                A.CallTo(
+                    () => progressDataService.UpdateProgressSupervisorAndCompleteByDate(
+                        reusableProgressRecord.ProgressId,
+                        A<int>._,
+                        expectedFutureDate
+                    )
+                ).MustHaveHappened();
+            }
+        }
+
+        [Test]
+        public void EnrolDelegateOnGroupCourses_sends_email_on_successful_enrolment_with_correct_process_name()
+        {
+            // Given
+            SetupEnrolProcessFakes(
+                GenericNewProgressId,
+                GenericRelatedTutorialId,
+                reusableGroupCourse
+            );
+
+            // When
+            groupsService.EnrolDelegateOnGroupCourses(
+                reusableDelegateDetails,
+                reusableAccountDetailsData,
+                1,
+                "baseUrl"
+            );
+
+            // Then
+            using (new AssertionScope())
+            {
+                A.CallTo(
+                        () => emailService.ScheduleEmail(A<Email>._, "AddDelegateToGroup_Refactor", null)
+                    )
+                    .MustHaveHappened();
+            }
+        }
+
+        [Test]
+        public void EnrolDelegateOnGroupCourses_sends_correct_email_with_no_CompleteByDate()

```

Comment:
Strange. They both pass for me when run individually, as a file, and when running all tests. The Jenkins builds from when I first made the commits all seemed to have passed. Not sure why it would be failing now for you, but I'll have a further investigation.

Reply:
Yeah I've got no idea why this would be failing for you. I would normally suspect subtle differences in our test databases, but all data service queries here have been mocked. Is there any useful information about why it failed?

Reply:
They work now. 🤷